src/noise.ts

```diff
@@ -76,29 +97,83 @@ export class Noise implements NoiseConnection {
     };
   }
 
-  private async performHandshake(
-    connection: WrappedConnection,
-    isInitiator: boolean,
-    libp2pPublicKey: bytes,
-    remotePeer: PeerId,
-  ): Promise<Handshake> {
-    const prologue = Buffer.from(this.protocol);
-    const handshake = new Handshake(isInitiator, this.privateKey, libp2pPublicKey, prologue, this.staticKeys, connection, remotePeer);
+  /**
+   * If Noise pipes supported, tries IK handshake first with XX as fallback if it fails.
+   * If remote peer static key is unknown, use XX.
+   * @param connection
+   * @param isInitiator
+   * @param libp2pPublicKey
+   * @param remotePeer
+   */
+  private async performHandshake(params: HandshakeParams): Promise<IHandshake> {
+    // TODO: Implement noise pipes
+
+    if (false) {
+      let IKhandshake;
+      try {
+        IKhandshake = await this.performIKHandshake(params);
+        return IKhandshake;
+      } catch (e) {
+        // XX fallback
+        const ephemeralKeys = IKhandshake.getRemoteEphemeralKeys();
```

Comment:
but IKhandshake will be null in case of exception

Reply:
Not necessarily. Will see when IK is implemented.

Reply:
according to current code, in case of exception in `performIKHandshake` IKhandshake will be null in catch clause

Reply:
? The whole code for that is TODO

Reply:
? The whole code for that is under TODO